### PR TITLE
Handle signals and exit cleanly.

### PR DESCRIPTION
**Pull request by alokmenghrajani**
- [ ] @csstaub
- [ ] @mcpherrinm

### CODE REVIEW

File: keywhizfs/main.go [hunk 1]
```diff
@@ -92,6 +93,15 @@ func main() {
 		log.Fatalf("Mount fail: %v\n", err)
 	}
 
+	// Catch SIGINT and SIGKILL and exit cleanly.
+	c := make(chan os.Signal, 1)
+	signal.Notify(c, os.Interrupt, os.Kill)
```
Comment (csstaub) [inline keywhizfs/main.go]: SIGKILL can't be caught or ignored.

File: keywhizfs/main.go [hunk 2]
```diff
@@ -92,6 +93,15 @@ func main() {
...
+	go func() {
+		<-c
+		fmt.Fprintf(os.Stdout, "got signal, unmounting!")
```
Comment (csstaub) [inline keywhizfs/main.go]: Print actual signal you received here?

Comment (csstaub) [inline keywhizfs/main.go]: Also, why print and not log? Why stdout not stderr?

Comment (alokmenghrajani) [inline keywhizfs/main.go]: I was debugging.

**Comment by csstaub** [comment]
Failing build? Otherwise :+1:

**Comment by alokmenghrajani** [comment]
No idea why the build failed. Re-building the same sha.

**Pull request merged**